assets/app/features/upgrade/controllers/upgrade-landing.controller.js

```diff
@@ -86,6 +88,24 @@
                     });
                 });
             });
+
+            // adjust UI state to backend status
+            upgradeFactory.getStatus()
+                .then(
+                    function (response) {
+                        // skz: This part is disabled on purpose. There's no easy way to restore complete checks state
+                        // without running the checks again so it's better to leave this to the user.
+                        //vm.prechecks.running = response.data.steps.upgrade_prechecks.status === STEP_STATES.running;
+                        //vm.prechecks.completed = response.data.steps.upgrade_prechecks.status === STEP_STATES.passed;
+
+                        vm.prepare.running = response.data.steps.upgrade_prepare.status === STEP_STATES.running;
```

Comment:
I thought you mean just extracting this code into a local function. If you want to make it somehow reusable then it  makes more sense. I'll think about it.